components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/DefaultIdentityProviderImpl.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * <p>
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.apimgt.core.impl;
+
+import feign.Response;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.carbon.apimgt.core.api.IdentityProvider;
+import org.wso2.carbon.apimgt.core.auth.SCIMServiceStub;
+import org.wso2.carbon.apimgt.core.auth.SCIMServiceStubFactory;
+import org.wso2.carbon.apimgt.core.auth.dto.SCIMUser;
+import org.wso2.carbon.apimgt.core.exception.APIManagementException;
+import org.wso2.carbon.apimgt.core.exception.ExceptionCodes;
+import org.wso2.carbon.apimgt.core.exception.IdentityProviderException;
+import org.wso2.carbon.apimgt.core.models.User;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * API Manager default implementation of {@link IdentityProvider}
+ */
+public class DefaultIdentityProviderImpl extends DefaultKeyManagerImpl implements IdentityProvider {
+
+    private static final Logger log = LoggerFactory.getLogger(DefaultIdentityProviderImpl.class);
+
+    private SCIMServiceStub scimServiceStub;
+    private static final String FILTER_PREFIX = "displayName Eq ";
+    private static final String HOME_EMAIL = "home";
+
+    public DefaultIdentityProviderImpl() throws APIManagementException {
+        scimServiceStub = SCIMServiceStubFactory.getSCIMServiceStub();
+    }
+
+    protected DefaultIdentityProviderImpl(SCIMServiceStub scimServiceStub) {
+        this.scimServiceStub = scimServiceStub;
+    }
+
+    @Override
+    public List<String> getRolesOfUser(String userId) throws IdentityProviderException {
+        List<String> roleNames = new ArrayList<>();
+        SCIMUser scimUser = scimServiceStub.getUser(userId);
+        if (scimUser != null) {
+            List<org.wso2.carbon.apimgt.core.auth.dto.SCIMUser.SCIMUserGroups> roles = scimUser.getGroups();
```

Comment:
Instead of using org.wso2.carbon.apimgt.core.auth.dto.SCIMUser.SCIMUserGroups Shall we import the class and use as SCIMUserGroups